File: mobile/lib/features/reservations/domain/reservation.dart
class Reservation {
  const Reservation({
    required this.id,
    required this.restaurantId,
    required this.restaurantName,
    required this.name,
    required this.guests,
    required this.startsAt,
    required this.durationMin,
    required this.status,
    this.userId,
    this.tableObjectId,
    this.tableName,
    this.phone,
    this.comment,
  });

  final int id;
  final int restaurantId;
  final String restaurantName;
  final int? userId;
  final int? tableObjectId;
  final String? tableName;
  final String name;
  final String? phone;
  final int guests;
  final DateTime startsAt;
  final int durationMin;
  final String status;
  final String? comment;

  bool get canCancel => status == 'requested' || status == 'confirmed';

  factory Reservation.fromJson(Map<String, dynamic> json) => Reservation(
    id: json['id'] as int,
    restaurantId: json['restaurant_id'] as int,
    restaurantName: json['restaurant_name'] as String? ?? '',
    userId: json['user_id'] as int?,
    tableObjectId: json['table_object_id'] as int?,
    tableName: json['table_name'] as String?,
    name: json['name'] as String,
    phone: json['phone'] as String?,
    guests: json['guests'] as int? ?? 2,
    startsAt: DateTime.parse(json['starts_at'] as String),
    durationMin: json['duration_min'] as int? ?? 90,
    status: json['status'] as String,
    comment: json['comment'] as String?,
  );
}

File: backend/app/schemas/reservation.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field, field_validator


class ReservationCreate(BaseModel):
    restaurant_id: int
    table_object_id: int | None = None
    name: str = Field(min_length=1, max_length=80)
    phone: str | None = Field(default=None, max_length=30)
    guests: int = Field(default=2, ge=1, le=20)
    starts_at: datetime
    duration_min: int = Field(default=90, ge=60, le=180)
    comment: str | None = Field(default=None, max_length=500)

    @field_validator("name")
    @classmethod
    def _name(cls, value: str) -> str:
        text = value.strip()
        if not text:
            raise ValueError("Name is empty")
        return text

    @field_validator("phone")
    @classmethod
    def _phone(cls, value: str | None) -> str | None:
        if value is None:
            return None
        text = value.strip()
        return text or None


class ReservationUpdate(BaseModel):
    status: str | None = Field(
        default=None,
        pattern="^(requested|confirmed|seated|cancelled|no_show)$",
    )
    table_object_id: int | None = None
    guests: int | None = Field(default=None, ge=1, le=20)


class ReservationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    restaurant_id: int
    restaurant_name: str = ""
    user_id: int | None
    table_object_id: int | None
    table_name: str | None = None
    name: str
    phone: str | None
    guests: int
    starts_at: datetime
    duration_min: int
    status: str
    comment: str | None
    created_at: datetime


class TableOut(BaseModel):
    id: int
    name: str
    capacity: int | None
    status: str
    floor_name: str
    kind: str
